azurelinuxagent/common/osutil/default.py

```diff
@@ -1411,6 +1412,30 @@ def get_total_cpu_ticks_since_boot():
                     break
         return system_cpu
 
+    @staticmethod
+    def get_used_and_available_system_memory():
+        """
+        Get the contents of free -b in bytes.
+        # free -b
+        #              total        used        free      shared  buff/cache   available
+        # Mem:     8340144128   619352064  5236809728     1499136  2483982336  7426314240
+        # Swap:             0           0           0
+
+        :return: used and available memory in megabytes
+        """
+        used_mem = available_mem = 0
+        free_cmd = ["free", "-b"]
+        try:
+            memory = shellutil.run_command(free_cmd)
+            for line in memory.split("\n"):
+                if ALL_MEMS_REGEX.match(line):
+                    mems = line.split()
+                    used_mem = int(mems[2])
+                    available_mem = int(mems[6])  # see "man free" for a description of these fields
+        except CommandError as e:
+            logger.warn("Cannot get the memory table. {0} failed: {1}", ustr(free_cmd), ustr(e))
```

Comment:
if this fails continuously we would be producing a warning on each period

you can let the exception bubble up and handle it on the caller (the periodic task), which can log the error once (until it succeeds again).